internal/proxy/readiness.go

```diff
@@ -1,63 +1,265 @@
 package proxy
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/kubeflow/model-registry/internal/datastore"
 	"github.com/kubeflow/model-registry/internal/db"
+	"github.com/kubeflow/model-registry/pkg/api"
 )
 
-// ReadinessHandler is a readiness probe that requires schema_migrations.dirty to be false before allowing traffic.
-func ReadinessHandler(datastore datastore.Datastore) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// skip embedmd check for mlmd datastore
-		if datastore.Type != "embedmd" {
-			w.WriteHeader(http.StatusOK)
-			_, _ = w.Write([]byte("OK"))
-			return
-		}
+// HealthCheck represents a single health check
+type HealthCheck struct {
+	Name    string
+	Status  string
+	Message string
+	Details map[string]interface{}
+}
+
+// HealthStatus represents the overall health status
+type HealthStatus struct {
+	Status string                 `json:"status"`
+	Checks map[string]HealthCheck `json:"checks"`
+}
+
+// HealthChecker defines the interface for health checks
+type HealthChecker interface {
+	Check() HealthCheck
+}
+
+// DatabaseHealthChecker checks database connectivity and schema state
+type DatabaseHealthChecker struct {
+	datastore datastore.Datastore
+}
+
+func NewDatabaseHealthChecker(datastore datastore.Datastore) *DatabaseHealthChecker {
+	return &DatabaseHealthChecker{
+		datastore: datastore,
+	}
+}
+
+func (d *DatabaseHealthChecker) Check() HealthCheck {
+	check := HealthCheck{
+		Name:    "database",
```

Comment:
Could this be a list of consts? Same thing for other string literals like Status.